internal/server/util/jwt_service.go

```diff
@@ -0,0 +1,208 @@
+package util
+
+import (
+	"crypto/x509"
+	stdjson "encoding/json" //nolint: depguard
+	"encoding/pem"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/google/uuid"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type IJwtService interface {
+	JwtPublicKey() interface{}
+	ParseAuthorization(authorization []string) (*JWTClaims, error)
+}
+
+type JWTClaims struct {
+	jwt.RegisteredClaims
+	Oid        string `json:"oid"`
+	OrgName    string `json:"org_name"`
+	OrgOwner   bool   `json:"user_is_org_owner"`
+	FeatureSet string `json:"feature_set"`
+}
+
+type JwtService struct {
+	jwtPublicKey interface{}
+}
+
+type NewJwtServiceOpts struct {
+	JwtPublicKey string
+}
+
+func New(opts NewJwtServiceOpts) (*JwtService, error) {
+	// parsing in case of bad env variable parsing
+	parsedStr := strings.ReplaceAll(opts.JwtPublicKey, "\\n", "\n")
+
+	raw, rest := pem.Decode([]byte(parsedStr))
+	if len(rest) > 0 {
+		return nil, errors.New("rest found, Jwt key invalid")
+	}
+	rsaPublicKey, err := x509.ParsePKIXPublicKey(raw.Bytes)
+	if err != nil {
+		return nil, err
+	}
+	return &JwtService{
+		jwtPublicKey: rsaPublicKey,
+	}, nil
+}
+
+func (s *JwtService) JwtPublicKey() interface{} {
+	return s.jwtPublicKey
+}
+
+var errNoClaim = errors.New("unable to parse claims")
+
+func (s *JwtService) ParseAuthorization(tokenString string) (*JWTClaims, error) {
+	tokenSlice := strings.Split(tokenString, " ")
+	if len(tokenSlice) < 2 { //nolint: gomnd
```

Comment:
```suggestion
	if len(tokenSlice) < 2  || tokenSlice[0] != "Bearer" { //nolint: gomnd
```

Could also add the other check to make the code more concise.

Reply:
Done